svgdigitizer/electrochemistry/electrolyte.py

```diff
@@ -0,0 +1,196 @@
+    'JUJWROOIHBZHMG-UHFFFAOYSA-N': {'name': 'pyridine', 'components': [Acid(pKa=[5.23], charge=1)]},  # CRC handbook of chemistry and physics 2017
+    'QGZKDVFQNNGYKY-UHFFFAOYSA-N': {'name': 'NH3', 'components': [Acid(pKa=[9.25], charge=1)]},  # CRC handbook of chemistry and physics 2017
+    'BVKZGUZCCUSVTD-UHFFFAOYSA-N': {'name': 'H2CO3', 'components': [Acid(pKa=[6.35, 10.33], charge=0)]},  # CRC handbook of chemistry and physics 2017
+    'HEMHJVSKTPXQMS-UHFFFAOYSA-M': {'name': 'NaOH', 'components': [Neutral(charge=1)]},
+    'FAPWRFPIFSIZLT-UHFFFAOYSA-M': {'name': 'NaCl', 'components': [Acid(pKa=[-6.2], charge=0), Neutral(charge=1)]},
+    'PMZURENOXWZQFD-UHFFFAOYSA-L': {'name': 'Na2SO4', 'components': [Acid(pKa=[-3, 1.9], charge=0), Neutral(charge=2)]},
+    'RYFMWSXOAZQYPI-UHFFFAOYSA-K': {'name': 'Na3PO4', 'components': [Acid(pKa=[2.16, 7.21, 12.32], charge=0), Neutral(charge=3)]},
+    'BNIILDVGGAEEIG-UHFFFAOYSA-L': {'name': 'Na2HPO4', 'components': [Acid(pKa=[2.16, 7.21, 12.32], charge=0), Neutral(charge=2)]},
+    'AJPJDKMHJJGVTQ-UHFFFAOYSA-M': {'name': 'NaH2PO4', 'components': [Acid(pKa=[2.16, 7.21, 12.32], charge=0), Neutral(charge=1)]},
+    }
+
+
+class Electrolyte():
+    r"""
+    Electrolyte
+    """
+    def __init__(self, electrolyte):
+        self.electrolyte = electrolyte
+
+    @property
+    @cache
+    def pH(self):
+        r"""
+        pH of the electrolyte.
+        """
+        try:
+            return self.electrolyte['pH']
+        except KeyError:
+            return self.estimate_pH()
+
+    def estimate_pH(self):
+        r"""
+        Estimates the pH of aquoues electrolytes.
+
+        EXAMPLES::
+
+            CID can be retrieved with InChIKey::
+            >>> from svgdigitizer.electrochemistry.electrolyte import Electrolyte
+            >>> electrolyte_dict = {"type": "aq", "components": [{"name": "H2SO4","type": "acid", "concentration": {"value": 0.1, "unit": "M"}}, {"name": "Na2SO4","type": "salt", "concentration": {"value": 0.1, "unit": "M"}}]}
+            >>> electrolyte = Electrolyte(electrolyte_dict)
+            >>> electrolyte.pH
+            1.3538681030273414
+        """
+
+        if self.electrolyte['type'] != 'aq':
+            raise NotImplementedError("pH calculations are only implemented for aqueous solutions!")
+        acids_bases = []
+        for i in self.electrolyte['components']:
+            if not i['type'] == 'solvent':
+                InChIKey = self.lookup_chemical_ids_from_name(i['name'])["InChIKey"]
+                temp = self.get_pKa(InChIKey)
+                for e in temp:
+                    q = i['concentration']['value'] * normalize_unit(i['concentration']['unit'])
+                    e.conc = q.to("mol / l").value
+                acids_bases.extend(temp)
+
+        system = System(*acids_bases)
+        system.pHsolve()
+
+        return system.pH
+
+    @classmethod
+    def lookup_CID_from_InChI(cls, InChIKey=None, InChI=None):
```

Comment:
I am trying to understand how the lookup methods invoke each other to see if this can be simplified:
```
estimate_pH invokes lookup_chemical_ids_from_name and get_pKa.
get_pKa invokes lookup_CID_from_InChI, lookup_pKa.
```

So, this method is never called with an `InChI` actually.

However, I think it's

Reply:
However, I personally find `InChI` easier to understand so maybe we can remove the `InChIKey` everywhere? (Or remove `InChI` everywhere.)

Reply:
Why are these methods called `lookup`? Could we call them all `get` instead?

Reply:
From the invocation scheme above, it seems that this method can be removed completely.
`estimate_pH` calls `lookup_chemical_ids` so we already know the `CID`. There does not seem to be a reason to recover it from the `InChI` again later. I am not sure what's the best way to resolve this. We could either use the `chemical name` everywhere, or the full set of `chemical ids`.